src/init.cpp

```diff
@@ -1563,6 +1551,9 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
     RegisterValidationInterface(node.peerman.get());
 
     // ********************************************************* Step 8: start indexers
+
+    // If reindex-chainstate was specified, delay syncing indexes until ThreadImport has reindexed the chain
+    g_indexes_ready_to_sync = !fReindexChainState;
```

Comment:
In commit "index: Enable reindex-chainstate with active indexes" (91947e4387d98290d39c9201f85a9aefa235a4d1)

This is a pedantic, academic suggestion, but should consider writing this as:

```c++
if (!fReindexChainState) g_indexes_ready_to_sync = true;
```

To make this part of the init function idempotent, and also make it clearer that index code only needs to handle `g_indexes_ready_to_sync` transitioning from false to true, never the other way around.

Reply:
Done with the latest push.